File: Problemas/cpp/bancos-con-herencia/java/Divisa.java
class Divisa extends Cuenta
{
	private double	ratio;
	private Cuenta	otra;

	public Divisa(double _ratio, Cuenta _otra)
	{
		super(0);
		ratio = _ratio;
		otra  = _otra;
	};

	// Implementa el calculo del valor
	@Override
	public double valor(int t)
	{
		return ratio*otra.valor(t);
	}

	// Redefine la forma de actualizarse
	@Override  //Cuando ha hecho el ej no ha puesto override en ningun sitio
	public void actualiza()
	{
		otra.actualiza();
	}
}

File: Problemas/cpp/bancos-con-herencia/c++/divisa.cc
#include "divisa.h"

Divisa::Divisa(double _ratio, Cuenta& _otra)
	: Cuenta(0), ratio(_ratio), otra(_otra)
{
}

double Divisa::valor(int t) const
{
	return ratio*otra.valor(t);
}

void Divisa::actualiza()
{
	otra.actualiza();
}
